src/microbe_stage/systems/StrainSystem.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Components;
+using DefaultEcs;
+using DefaultEcs.System;
+using DefaultEcs.Threading;
+using Godot;
+using World = DefaultEcs.World;
+
+[With(typeof(OrganelleContainer))]
+[With(typeof(StrainAffected))]
+[ReadsComponent(typeof(OrganelleContainer))]
+[ReadsComponent(typeof(MicrobeControl))]
+public sealed class StrainSystem : AEntitySetSystem<float>
+{
+    public StrainSystem(World world, IParallelRunner runner) : base(world, runner)
+    {
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        ref var strain = ref entity.Get<StrainAffected>();
+        ref var organelles = ref entity.Get<OrganelleContainer>();
+
+        if (strain.IsUnderStrain)
+        {
+            var strainIncrease = Constants.SPRINTING_STRAIN_INCREASE_PER_SECOND * delta;
+            strainIncrease += organelles.HexCount * Constants.SPRINTING_STRAIN_INCREASE_PER_HEX;
+
+            strain.CurrentStrain += strainIncrease;
+
+            if (strain.CurrentStrain > Constants.MAX_STRAIN_PER_ENTITY)
+                strain.CurrentStrain = Constants.MAX_STRAIN_PER_ENTITY;
+
+            strain.StrainDecreaseCooldown = Constants.STRAIN_DECREASE_COOLDOWN_SECONDS;
+        }
+        else
+        {
+            if (strain.StrainDecreaseCooldown <= Mathf.Epsilon)
```

Comment:
Can't this variable go negative as well? It should be fine to just compare against zero in that case as I think it'll be more performant than comparing against an epsilon value unnecessarily.